homeassistant/components/pi_hole/config_flow.py

```diff
@@ -0,0 +1,150 @@
+"""Config flow to configure the Pi-hole integration."""
+import logging
+
+from hole import Hole
+from hole.exceptions import HoleError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.pi_hole.const import (  # pylint: disable=unused-import
+    CONF_LOCATION,
+    DEFAULT_LOCATION,
+    DEFAULT_NAME,
+    DEFAULT_SSL,
+    DEFAULT_VERIFY_SSL,
+    DOMAIN,
+)
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DuplicatedNameException(Exception):
+    """The name is already existed."""
+
+
+class PiHoleFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Pi-hole config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        return await self.async_step_init(user_input)
+
+    async def async_step_import(self, user_input=None):
+        """Handle a flow initiated by import."""
+        return await self.async_step_init(user_input, is_import=True)
+
+    async def async_step_init(self, user_input, is_import=False):
+        """Handle init step of a flow."""
+        errors = {}
+
+        if user_input is not None:
+            host = (
+                user_input[CONF_HOST]
+                if is_import
+                else f"{user_input[CONF_HOST]}:{user_input[CONF_PORT]}"
+            )
+            name = user_input[CONF_NAME]
+            location = user_input[CONF_LOCATION]
+            tls = user_input[CONF_SSL]
+            verify_tls = user_input[CONF_VERIFY_SSL]
+            api_token = user_input.get(CONF_API_KEY)
+            endpoint = f"{host}/{location}"
```

Comment:
I couldn't get this to connect
`2020-05-10 21:49:32 ERROR (MainThread) [hole] Can not load data from *hole: 192.168.208.3:80`

I'm not sure what is going on.  I'll out of time to do debugging though

Reply:
Try access `http://192.168.208.3/admin/api.php` directly from your browser and see what you can get.

Reply:
```{"domains_being_blocked":82768,"dns_queries_today":147652,"ads_blocked_today":2943,"ads_percentage_today":1.9932,"unique_domains":3312,"queries_forwarded":68857,"queries_cached":75852,"clients_ever_seen":112,"unique_clients":110,"dns_queries_all_types":147652,"reply_NODATA":47,"reply_NXDOMAIN":50,"reply_CNAME":474,"reply_IP":368,"privacy_level":0,"status":"enabled","gravity_last_updated":{"file_exists":true,"absolute":1589147094,"relative":{"days":"0","hours":"00","minutes":"35"}}}```

Reply:
Disregard, its a routing issue from my HA instance to the pihole
curl: (7) Failed to connect to 192.168.208.3 port 80: No route to host

Reply:
The library does exactly the same thing. Maybe you should try to log the original exception [here](https://github.com/fabaff/python-hole/blob/b6d085a015f111940d57662052c1d8e07902dda2/hole/__init__.py#L52) to see what's actually going on.